Code/Framework/AzToolsFramework/AzToolsFramework/API/EditorAssetSystemAPI.h

```diff
@@ -127,6 +127,14 @@ namespace AzToolsFramework
             * returns false if it cannot find the source, true otherwise.
             */
             virtual bool GetAssetsProducedBySourceUUID(const AZ::Uuid& sourceUuid, AZStd::vector<AZ::Data::AssetInfo>& productsAssetInfo) = 0;
+
+            /** This will cause the file to reprocess the next time it changes, even if it's identical to what it was before.
+            * This is useful for in-editor tools that save source files. A content creator expects that, every time they save a file,
+            * the file will be processed, even if nothing has actually changed, so they will sometimes save a file specifically to
+            * force the asset to reprocess.
+            * @return true if the fingerprint was cleared, false if not.
+            */
+            virtual bool ClearFingerprintForAsset(const AZStd::string& sourcePath) = 0;
```

Comment:
this might be good to plug into the AP as well to "reprocess" the asset, right?

Reply:
The AP already has an existing, different workflow for reprocessing assets. Right click on an asset and you can reprocess it.

This is different because the Editor was sending the command, it wasn't happening in AP directly, and in this case I didn't want to immediately reprocess, I wanted to flag the file in a way it would reprocess the next time it saved, to avoid race conditions if I requested a reprocess after saving the file, to avoid having to have the Editor code check if the fingerprint changed or not.